dask/dataframe/shuffle.py

```diff
@@ -24,6 +25,196 @@
 logger = logging.getLogger(__name__)
 
 
+class ShuffleStage(Layer):
+    """Shuffle HighLevelGraph Layer
+
+    High-level graph layer corresponding to a single stage of
+    an inter-partition shuffle operation.
+
+    Stage: (shuffle-group) -> (shuffle-split) -> (shuffle-join)
+
+    Parameters
+    ----------
+    name : str
+        Name of new (partially) shuffled collection.
+    column : str or list of str
+        Column(s) to be used to map rows to output partitions (by hashing).
+    inputs : list of tuples
+        Each tuple dictates the data movement for a specific partition.
+    stage : int
+        Index of the current shuffle stage.
+    npartitions : int
+        Number of output partitions for the full (multi-stage) shuffle.
+    n : int
+        Number of partitions in the original (un-shuffled) DataFrame.
+    k : int
+        A partition is split into this many groups during each stage.
+    ignore_index: bool, default False
+        Ignore index during shuffle.  If ``True``, performance may improve,
+        but index values will not be preserved.
+    dep_name : str
+        Name of input collection.
+    dep_meta : pd.DataFrame-like object
+        Empty metadata of input collection.
+    parts_out : list of int (optional)
+        List of required output-partition indices.
+    """
+
+    def __init__(
+        self,
+        name,
+        column,
+        inputs,
+        stage,
+        npartitions,
+        n,
+        k,
```

Comment:
descriptive parameter names here would be nice. Suggestions: `n_input_partitions`, and `n_groups_per_partition`

Reply:
Right - We should definitely adopt better names. How about `n` -> `npartitions_input` and `k` -> `nsplits`?